presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.teradata.tempto.AfterTestWithContext;
+import com.teradata.tempto.BeforeTestWithContext;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.query.QueryResult;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Response;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.tests.TestGroups.CANCEL_QUERY;
+import static com.google.common.base.Preconditions.checkState;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.lang.System.nanoTime;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static org.testng.Assert.fail;
+
+public class SqlCancelTests
+        extends ProductTest
+{
+    private ExecutorService executor;
+    private QueryCanceller queryCanceller;
+
+    @Inject
+    @Named("databases.presto.server_address")
+    private String serverAddress;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        executor = newFixedThreadPool(2);
+        queryCanceller = new QueryCanceller(serverAddress);
+    }
+
+    @AfterTestWithContext
+    public void cleanUp()
+    {
+        if (executor != null) {
+            executor.shutdownNow();
+        }
+        if (queryCanceller != null) {
+            queryCanceller.close();
+        }
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60 * 1000L)
```

Comment:
if this test is only in this one group, will it be run on travis?

`s/60 * 1000L/60_000/`

Reply:
yes. Travis run all, but some groups that are excluded.